internal/domain/tenant/model.go

```diff
@@ -9,11 +10,44 @@ import (
 
 // Tenant represents an organization or group within the system.
 type Tenant struct {
-	ID        string       `json:"id"`
-	Name      string       `json:"name"`
-	Status    types.Status `json:"status"`
-	CreatedAt time.Time    `json:"created_at"`
-	UpdatedAt time.Time    `json:"updated_at"`
+	ID             string         `json:"id"`
+	Name           string         `json:"name"`
+	Status         types.Status   `json:"status"`
+	BillingDetails BillingDetails `json:"billing_details"`
+	CreatedAt      time.Time      `json:"created_at"`
+	UpdatedAt      time.Time      `json:"updated_at"`
+}
+
+// Address represents a physical address
+type Address struct {
+	Line1      string `json:"address_line1"`
+	Line2      string `json:"address_line2"`
+	City       string `json:"address_city"`
+	State      string `json:"address_state"`
+	PostalCode string `json:"address_postal_code"`
+	Country    string `json:"address_country"`
+}
+
+// BillingDetails contains tenant billing information
+type BillingDetails struct {
+	Email     string  `json:"email"`
+	HelpEmail string  `json:"help_email"`
+	Phone     string  `json:"phone"`
+	Address   Address `json:"address"`
+}
+
+func (b *BillingDetails) ToMap() map[string]interface{} {
```

Comment:
why is this required?

Reply:
We do not the map string interface conversions. Refer the handling of `TransformQuantity` in price domain/schema to understand how we map struct as json fields in db